### PR TITLE
Add file request support

### PR DESCRIPTION
**Pull request by sujaygarlanka**

### CODE REVIEW

File: doc/file_requests.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [Get a File Request's Information](#get-a-file-requests-information)
+- [Copy a File Request's Information](#copy-a-file-requests-information)
+- [Update a File Request's Information](#update-a-file-requests-information)
+- [Delete a File Request](#delete-a-file-request)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+Get a File Requests's Information
```
Comment (swfree) [inline doc/file_requests.md:16]: minor: "File Request's"

File: src/main/java/com/box/sdk/BoxFileRequest.java [hunk 2]
```diff
@@ -0,0 +1,434 @@
...
+        BoxJSONResponse response = (BoxJSONResponse) request.send();
+        JsonObject responseJSON = JsonObject.readFrom(response.getJSON());
+        return new Info(responseJSON);
+    }
+
+    /**
+     * Copies this file request that is already present on one folder, and applies it to another folder.
+     *
+     * <p>The only fields that will be copied are the ones that have been modified locally. For example, the following
+     * code won't copy any information (or even send a network request) since none of the info's fields were
```
Comment (swfree) [inline src/main/java/com/box/sdk/BoxFileRequest.java:52]: Just confirming - so the user will always need to make a change to the info, like the description or status? What happens if they want to copy a file request as-is, without any changes?

File: src/main/java/com/box/sdk/BoxFileRequest.java [hunk 3]
```diff
@@ -0,0 +1,434 @@
...
+     */
+    public void delete() {
+        URL url = FILE_REQUEST_URL_TEMPLATE.build(this.getAPI().getBaseURL(), this.getID());
+        BoxAPIRequest request = new BoxAPIRequest(this.getAPI(), url, "DELETE");
+        BoxAPIResponse response = request.send();
+        response.disconnect();
+    }
+
+    /**
+     * Contains information about a BoxFileRequst.
```
Comment (swfree) [inline src/main/java/com/box/sdk/BoxFileRequest.java:112]: minor typo: BoxFileRequest

File: src/main/java/com/box/sdk/BoxFileRequest.java [hunk 4]
```diff
@@ -0,0 +1,434 @@
...
+                    }
+                }
+            } catch (Exception e) {
+                throw new BoxDeserializationException(memberName, value.toString(), e);
+            }
+        }
+    }
+
+    /**
+     * Enumerates the possible download states of a zip.
```
Comment (swfree) [inline src/main/java/com/box/sdk/BoxFileRequest.java:407]: Should this say "possible states of a file request"?

File: src/main/java/com/box/sdk/BoxFileRequest.java [hunk 5]
```diff
@@ -0,0 +1,434 @@
...
+     */
+    public enum Status {
+        /**
+         * Succeeded in downloading.
+         */
+        ACTIVE("active"),
+
+        /**
+         * Downloading in progress.
```
Comment (swfree) [inline src/main/java/com/box/sdk/BoxFileRequest.java:416]: should this mention something like in the docs - "An inactive file request will no longer accept new submissions, and any visitor to the file request URL will receive a HTTP 404 status code."

File: src/main/java/com/box/sdk/BoxFileRequest.java [hunk 6]
```diff
@@ -0,0 +1,434 @@
...
+            }
+        }
+    }
+
+    /**
+     * Enumerates the possible download states of a zip.
+     */
+    public enum Status {
+        /**
+         * Succeeded in downloading.
```
Comment (swfree) [inline src/main/java/com/box/sdk/BoxFileRequest.java:411]: Should this say something like "File request is active"

**Review by swfree — APPROVED** [review]

**Review by PJSimon — APPROVED** [review]

**Pull request merged**